provider/cloudflare/cloudflare.go

```diff
@@ -553,10 +555,10 @@ func (p *CloudFlareProvider) getRecordID(records []cloudflare.DNSRecord, record
 	return ""
 }
 
-func (p *CloudFlareProvider) getCustomHostnameOrigin(chs []cloudflare.CustomHostname, hostname string) (string, string) {
-	for _, zoneCh := range chs {
-		if zoneCh.Hostname == hostname {
-			return zoneCh.ID, zoneCh.CustomOriginServer
+func getCustomHostnameIdWithOrigin(chs []cloudflare.CustomHostname, hostname string) (string, string) {
```

Comment:
```suggestion
func getCustomHostnameIdWithOrigin(chs []cloudflare.CustomHostname, hostname string) (cloudflare.CustomHostname, error) {
```
Wdyt about:
1. return the whole object, so it's possible after to get ID, Origin and/or other fields
2. be explicit about not found, by returning a typed error

Reply:
thank you for the suggestion